src/databricks/labs/ucx/source_code/python_libraries.py

```diff
@@ -46,41 +58,81 @@ def _temporary_virtual_environment(self):
 
-    def _install_library(self, path_lookup: PathLookup, library: Path) -> list[DependencyProblem]:
+    def _install_library(
+        self, path_lookup: PathLookup, *libraries: str, installation_arguments: list[str]
+    ) -> list[DependencyProblem]:
         """Pip install library and augment path look-up to resolve the library at import"""
         path_lookup.append_path(self._temporary_virtual_environment)
+        resolved_libraries, resolved_installation_arguments = self._resolve_libraries(
+            path_lookup, *libraries, installation_arguments=installation_arguments
+        )
+        if libraries[0].endswith(".egg"):
+            return self._install_egg(*resolved_libraries)
+        return self._install_pip(*resolved_libraries, installation_arguments=resolved_installation_arguments)
 
+    @staticmethod
+    def _resolve_libraries(
+        path_lookup: PathLookup, *libraries: str, installation_arguments: list[str]
+    ) -> tuple[list[str], list[str]]:
         # Resolve relative pip installs from notebooks: %pip install ../../foo.whl
-        maybe_library = path_lookup.resolve(library)
-        if maybe_library is not None:
-            library = maybe_library
-
-        if library.suffix == ".egg":
-            return self._install_egg(library)
-        return self._install_pip(library)
-
-    def _install_pip(self, library: Path) -> list[DependencyProblem]:
-        return_code, stdout, stderr = self._runner(f"pip install {library} -t {self._temporary_virtual_environment}")
-        logger.debug(f"pip output:\n{stdout}\n{stderr}")
-        if return_code != 0:
-            problem = DependencyProblem("library-install-failed", f"Failed to install {library}: {stderr}")
-            return [problem]
-        return []
-
-    def _install_egg(self, library: Path) -> list[DependencyProblem]:
-        """Install egss using easy_install.
+        libs, args = [], installation_arguments.copy()
+        for library in libraries:
+            maybe_library = path_lookup.resolve(Path(library))
+            if maybe_library is None:
+                libs.append(library)
+            else:
+                args = [maybe_library.as_posix() if arg == library else arg for arg in args]
+                libs.append(maybe_library.as_posix())
+        return libs, args
+
+    def _install_pip(self, *libraries: str, installation_arguments: list[str]) -> list[DependencyProblem]:
+        problems = []
+        venv = self._temporary_virtual_environment
+        install_commands = []
+        if len(installation_arguments) == 0:
+            for library in libraries:
+                install_command = f"pip install {shlex.quote(library)} -t {venv}"
+                install_commands.append(install_command)
+        else:
+            # pip allows multiple target directories in its call, it uses the last one, thus the one added here
+            install_command = f"pip install {shlex.join(installation_arguments)} -t {venv}"
```

Comment:
`install_command = f"pip install {shlex.quote(library)} -t {venv}"` and `install_command = f"pip install {shlex.join(installation_arguments)} -t {venv}"` are the same

Reply:
They aren't always the same: i) installation arguments are empty for the pip libraries defined on a job and ii) the libraries don't contain the installation flags when those are defined in a `%pip` command